gitlab/v4/objects.py

```diff
@@ -1764,6 +1780,65 @@ def create(self, data, **kwargs):
         return CreateMixin.create(self, data, path=path, **kwargs)
 
 
+class ProjectPipelineScheduleVariable(SaveMixin, ObjectDeleteMixin, RESTObject):
+    _id_attr = 'key'
+
+
+class ProjectPipelineScheduleVariableManager(CRUDMixin, RESTManager):
+    _path = '/projects/%(project_id)s/pipeline_schedules/%(pipeline_schedule_id)s/variables'
+    _obj_cls = ProjectPipelineScheduleVariable
+    _from_parent_attrs = {'project_id': 'project_id',
+                          'pipeline_schedule_id' : 'id'}
+    _create_attrs = (('pipeline_schedule_id', 'key', 'value'), tuple())
+    _create_attrs = (('key', 'value'), tuple())
+
+    def list(self):
+        array = []
+        if 'variables' in self._parent._attrs:
+            for variable in self._parent._attrs['variables']:
+                schedule_variable = self._obj_cls(self, variable)
+                array.append(schedule_variable)
+        else:
+            obj = self._parent.manager.get(self._parent.id)
+            for variable in obj._attrs['variables']:
+                schedule_variable = self._obj_cls(self, variable)
+                array.append(schedule_variable)
+
+        return array
+
+
+class ProjectPipelineSchedule(RESTObject):
```

Comment:
GitLab supports updating and deleting these resources, so you need to add the `SaveMixin` and `ObjectDeleteMixin` here.